src/lib/containers/synapse_table_functions/SynapseTableCell.tsx

```diff
@@ -90,137 +76,157 @@ export const SynapseTableCell: React.FC<SynapseTableCellProps> = ({
-    if (isShort) {
-      return <p className={isBold}> {columnValue}</p>
-    } else {
+    case ColumnType.DATE:
       return (
-        <div className={isBold}>
-          <ElementWithTooltip
-            tooltipText={columnValue}
-            callbackFn={noop}
-            idForToolTip={`${colIndex}_${rowIndex}`}
-          >
-            <p className={isBold}> {columnValue}</p>
-          </ElementWithTooltip>
-        </div>
+        <p className={isBold}>
+          {new Date(Number(columnValue)).toLocaleString()}
+        </p>
       )
+
+    case ColumnType.USERID:
+      if (
+        Object.prototype.hasOwnProperty.call(mapUserIdToHeader, columnValue)
+      ) {
+        const { ownerId, userName } = mapUserIdToHeader[columnValue]
+        if (mapUserIdToHeader[columnValue].isIndividual === false) {
+          // isUserGroupHeader
+          const icon =
+            userName === AUTHENTICATED_USERS ? 'globe-americas' : 'users'
+          if (userName === AUTHENTICATED_USERS) {
+            return (
+              <span>
+                <FontAwesomeIcon icon={icon} /> All registered Synapse users
+              </span>
+            )
+          }
+          return (
+            <a
+              target="_blank"
+              rel="noopener noreferrer"
+              href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Team:${ownerId}`}
+            >
+              <FontAwesomeIcon icon={icon} /> {userName}
+            </a>
+          )
+        } else {
+          // isUserCard
+          return (
+            <UserCard
+              userProfile={mapUserIdToHeader[columnValue]}
+              preSignedURL={mapUserIdToHeader[columnValue].clientPreSignedURL}
+              size={'SMALL USER CARD'}
+            />
+          )
+        }
+      }
+      break
+    case ColumnType.STRING:
+    case ColumnType.DOUBLE:
+    case ColumnType.INTEGER:
+    case ColumnType.BOOLEAN:
+    case ColumnType.LINK:
+    case ColumnType.LARGETEXT: {
+      const isShort = isShortString(columnValue)
+      if (isShort) {
+        return <p className={isBold}> {columnValue}</p>
+      } else {
+        return (
+          <div className={isBold}>
+            <ElementWithTooltip
+              tooltipText={columnValue}
+              callbackFn={noop}
+              idForToolTip={`${columnName}_${rowIndex}`}
+            >
+              <p className={isBold}> {columnValue}</p>
+            </ElementWithTooltip>
+          </div>
+        )
+      }
     }
+    default:
+      console.warn(
+        `ColumnType ${columnType} has unspecified handler. Rendering the column value.`,
+      )
+      return <p className={isBold}>{columnValue}</p>
   }
+  // We can reach this if we don't get a mapping of IDs to entities or principals.
+  // TODO: If we don't have a id:data mapping, we should render a component that can fetch the required data, rather than breaking from the case.
+  return <p className={isBold}>{columnValue}</p>
```

Comment:
We will only be able to get here if we don't have the entity ID or user ID stored in the ID->data mappings that were passed as props. I'm guessing that this happens when we get a 403/404.

So my guess is that all of the cases where this can happen are 
* cache miss (these probably aren't happening)
* 403 error on an entity
* 404 error on an entity/user/team

It would make more sense to me to render a component that can fetch the data (so we don't have to worry about cache misses), and explicitly handle the 403/404 cases.

Since that would be a substantial change, I didn't do it here.